### PR TITLE
Pipenv dependency management

### PR DESCRIPTION
**Pull request by reinvantveer**
**Description of the change:**
Use [Pipenv](https://pipenv.pypa.io/en/latest/) - the Python packaging authority recommended way - as a dependency resolution managemente system and vulnerability checker to install Python dependencies. 

This PR is part of #4237 to work towards a more user-friendly way of doing reproducible local builds

**Motivation for the change:**
Python package dependency is best managed using a fit-for-purpose package manager that checks for dependency conflicts, separates wanted dependencies from required subdependencies and that is able to check for vulnerabilities. The proposed solution is fully backwards compatible.

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [x] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Comment by reinvantveer** [comment]
Note that the commit list is rather long simply because I rebased and merged from upstream, but the Python version pinning and Docker image pinning were already merged.

**Comment by reinvantveer** [comment]
Yay, passed!

File: images/ansible-operator/Dockerfile [hunk 1]
```diff
@@ -13,21 +13,20 @@ ENV HOME=/opt/ansible \
 # Install python dependencies
 # Ensure fresh metadata rather than cached metadata in the base by running
 # yum clean all && rm -rf /var/yum/cache/* first
```
Comment (estroz) [inline images/ansible-operator/Dockerfile:15]: This step still needs to be run

**Comment by estroz** [comment]
@reinvantveer looks like you need to rebase onto the latest master.

**Comment by reinvantveer** [comment]
> @reinvantveer looks like you need to rebase onto the latest master.

@estroz I'm in a spot of trouble here. I tried rebasing using `git pull --rebase` but now I apparently have 380 commits in this PR. How do I fix this? I'm a little in over my head here.

**Comment by reinvantveer** [comment]
I'll try in a new PR

File: images/ansible-operator/Dockerfile (hunk 1 above)
Comment (reinvantveer) [reply images/ansible-operator/Dockerfile:15]: Moved to #4494

**Pull request closed without merging**